### PR TITLE
Issue #467: Added method for updating a range of dates

### PR DESCRIPTION
**Pull request by pernilla**
I've created an issue #467 and this is a fix for that. It can be useful sometimes when wanting to update the calendar with a range of dates rather than a single date.

### CODE REVIEW

**Comment by edenman** [comment]
This behavior (only scrolling to the last specified date, and only if the last date was successfully selected) seems weird to me.  I'd think you'd actually want to scroll to the earliest selected date (to make it more likely you'd see the entire set of newly added/removed dates).

**Comment by pernilla** [comment]
I changed to first selected date, even though for me is not that important. If you create a review I can actually reply to your comment. As it is now I have to reply here.

File: library/src/main/java/com/squareup/timessquare/CalendarPickerView.java [hunk 1]
```diff
@@ -629,6 +629,55 @@ public boolean selectDate(Date date, boolean smoothScroll) {
...
+    boolean wasSelected = false;
+
+    int monthIndex = 1;
+
+    if (selectedDates != null) {
+      for (Date date : selectedDates) {
+        validateDate(date);
+
+        MonthCellWithMonthIndex monthCellWithMonthIndex = getMonthCellWithIndexByDate(date);
+        monthIndex = monthCellWithMonthIndex != null ? monthCellWithMonthIndex.monthIndex : 1;
```
[the remark this answers is not in the document]
Comment (pernilla) [reply library/src/main/java/com/squareup/timessquare/CalendarPickerView.java]: Fixed, waiting for this issue...

File: library/src/main/java/com/squareup/timessquare/CalendarPickerView.java [hunk 2]
```diff
@@ -629,6 +629,58 @@ public boolean selectDate(Date date, boolean smoothScroll) {
...
+
+        MonthCellWithMonthIndex monthCellWithMonthIndex = getMonthCellWithIndexByDate(date);
+
+        if (monthCellWithMonthIndex == null || !isDateSelectable(date)) {
+          return false;
+        }
+
+        monthIndex = monthCellWithMonthIndex.monthIndex;
+
+        if (!firstDateThatWasSelected) {
```
Comment (edenman) [inline library/src/main/java/com/squareup/timessquare/CalendarPickerView.java]: This doesn't look right.  You're only going to call `doSelectDate` on the first one that gets passed in.

Please add a test to `CalendarPickerViewTest` to make sure this all works as expected.

**Pull request closed without merging**